include/picongpu/particles/atomicPhysics/enums/IsProcess.hpp

```diff
@@ -107,11 +107,23 @@ namespace picongpu::particles::atomicPhysics::enums
         }
     };
 
+    //! processClass describing interaction with electric field
+    template<>
+    struct IsProcess<ProcessClassGroup::electricFieldBased>
+    {
+        static constexpr bool check(uint8_t const processClass)
+        {
+            if(processClass == u8(ProcessClass::fieldIonization))
```

Comment:
can't this be `if constexpr`?

Reply:
No because this function may also be called at runtime, in which case `processClass` is known at runtime only while `if constexpr` requires it condition to be known at compile time.
And we actually use this function to check if a given macro-ion belong to a certain process class at runtime, this behavior is therefore intended and required.